### DISCUSSION TITLE
How do I enable binary-caching when building with Visual Studio? (not CMake)

### DISCUSSION THREAD

**Discussion by kwsp** [microsoft/vcpkg#31176] [General]
I read https://learn.microsoft.com/en-us/vcpkg/users/binarycaching but its still not really clear.

For each visual studio project, it seems VCPKG automatically caches things in <project root>/vcpkg_installed, which is great, but after I checkout earlier commits (before I migrated a project to VCPKG), e.g. during a git-bisect, to previous commit when I used to manually manage deps, and then checkout HEAD again where `vcpkg.json` is used, VCPKG always rebuilds all deps, which takes ~30min for me.

The docs say
> Binary caching is enabled by default with a [files provider](https://learn.microsoft.com/en-us/vcpkg/users/binarycaching#files) at the first valid location of:

If its cached by default, why do my deps always rebuild after a git-bisect?

**Comment by dg0yt** [comment]
Caching is in APPDATA in windows. `vcpkg_installed` is the actual installation.

This caching is for whole artifacts, not individual translation units. The cache key includes all dependencies including central scripts. So bisect will lead to many cache misses if you never built these configurations. You might see more cache hits if you only vary individual ports.

**Reply by kwsp** [reply]
to clarify, it's after I finish a bisect, and go back to the current most up-to-date `vcpkg.json`, which has already been built before, when the rebuild is triggered for all deps. Is this normal? Thanks!

**Reply by dg0yt** [reply]
Not normal... If that configuration has been built before, the packages should be should be in the cache and be reused.
There are a few more input parameters, even the compiler binary. But it shouldn't vary in a short time span.
